server/command.go

```diff
@@ -121,6 +125,90 @@ func (p *Plugin) ExecuteCommand(c *plugin.Context, args *model.CommandArgs) (*mo
 		}
 		p.postCommandResponse(args, str.String())
 		return &model.CommandResponse{}, nil
+	case "set_channel_welcome":
+		if len(parameters) == 0 {
+			p.postCommandResponse(args, "`set_channel_welcome` command requires the message to be provided")
+			return &model.CommandResponse{}, nil
+		}
+
+		if channelInfo, appErr := p.API.GetChannel(args.ChannelId); appErr != nil {
+			errMsg := fmt.Sprintf(
+				"error occured while checking the type of the chanelId `%s`: `%s`",
+				args.ChannelId, err)
+			p.postCommandResponse(args, errMsg)
+			return &model.CommandResponse{}, nil
+		} else if channelInfo.Type == model.CHANNEL_PRIVATE {
+			p.postCommandResponse(
+				args, "welcome messages are not supported for direct channels")
+			return &model.CommandResponse{}, nil
+		}
```

Comment:
In general, returning earlier and not using `else` is favourable and the consistent style at Mattermost. So this would be rewritten to:
```suggestion
		channelInfo, appErr := p.API.GetChannel(args.ChannelId);
		if appErr != nil {
			errMsg := fmt.Sprintf(
				"error occured while checking the type of the chanelId `%s`: `%s`",
				args.ChannelId, err)
			p.postCommandResponse(args, errMsg)
			return &model.CommandResponse{}, nil
		} 
		
		if channelInfo.Type == model.CHANNEL_PRIVATE {
			p.postCommandResponse(
				args, "welcome messages are not supported for direct channels")
			return &model.CommandResponse{}, nil
		}
```